app/src/main/java/org/simple/clinic/editpatient/EditPatientEffectHandler.kt

```diff
@@ -172,6 +186,18 @@ class EditPatientEffectHandler @AssistedInject constructor(
     )
   }
 
+  private fun saveBangladeshNationalId(savePatientEffects: Observable<SavePatientEffect>): Observable<EditPatientEvent> {
+    return savePatientEffects
+        .flatMapCompletable {
+          if (it.bangladeshNationalId != null) {
+            patientRepository.saveBusinessId(it.bangladeshNationalId)
+          } else {
+            Completable.complete()
+          }
+        }
```

Comment:
I think we can replace this `if` condition to use a `filter`?

Reply:
Done.